source/beamline/SGM/monochromator/SGMEnergyTrajectory.cpp

```diff
@@ -0,0 +1,199 @@
+#include "SGMEnergyTrajectory.h"
+
+SGMEnergyTrajectory::SGMEnergyTrajectory(double startEnergy,
+                                         double endEnergy,
+                                         double time,
+                                         SGMGratingSupport::GratingTranslation gratingTranslation)
+{
+
+    startMonoInfo_ = new SGMMonochromatorInfo(startEnergy, gratingTranslation);
```

Comment:
Is there a design reason for new'ing the mono info's rather than just using them as stack variables?